test/mocha/function.js

```diff
@@ -46,22 +55,56 @@ describe("Function", function() {
         assert.deepEqual(
             [
                 destruct1.names[0].TYPE,
-                destruct1.names[0].name],
+                destruct1.names[0].name
+            ],
+            aAndB[0]);
+        assert.deepEqual(
+            [
+                destruct1.names[1].TYPE,
+                destruct1.names[1].name
+            ],
+            aAndB[1]);
+        assert.deepEqual(
+            [
+                destruct2.names[0].TYPE,
+                destruct2.names[0].name
+            ],
             aAndB[0]);
-
         assert.deepEqual(
             [
                 destruct2.names[1].names[0].TYPE,
                 destruct2.names[1].names[0].name
             ],
             aAndB[1]);
+        assert.strictEqual(typeof destruct3.names[0].key, "string");
+        assert.strictEqual(destruct3.names[0].key, "a");
+        assert.strictEqual(typeof destruct3.names[1].key, "string");
+        assert.strictEqual(destruct3.names[1].key, "b");
```

Comment:
Now arrow parameters binding properties are converted to AST_ObjectKeyVal (due to lib/ast.js with ex.shorthand). As described in the comments I prefer to get rid of the shorthand property and replace the whole node by a symbol node.